### PR TITLE
linux config: enable Creative Soundblaster DSP loading

### PR DESCRIPTION
**Pull request by Emantor**
###### Motivation for this change
Since we select everything as a module, snd_hda_codec_ca0132 is built as
well. DSP loading is not enabled by default, but without it the
soundcard produces timeouts within ALSA and does not emit sound.
Explicitly enable the firmware loading to ensure Soundblaster
Z/Zx/ZxR/Recon devices can be used with NixOS.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Emantor** [comment]
[Debian](https://salsa.debian.org/kernel-team/linux/blob/master/debian/config/config#L7787) and [Arch](https://git.archlinux.org/svntogit/packages.git/tree/trunk/config?h=packages/linux#n6764) enable this as well. I noticed this while switching to NixOS from Arch Linux.

**Comment by Emantor** [comment]
Takashi applied the [patch](https://git.kernel.org/pub/scm/linux/kernel/git/tiwai/sound.git/commit/?h=for-next&id=652bb5d8df4b3a79ed350db35cda12637e63efa7) to the sound tree to add `default y`. This can eventually be removed when all kernels default to `y`.

File: pkgs/os-specific/linux/kernel/common-config.nix [hunk 1]
```diff
@@ -244,6 +244,7 @@ let
       SND_HDA_RECONFIG    = yes; # Support reconfiguration of jack functions
       # Support configuring jack functions via fw mechanism at boot
       SND_HDA_PATCH_LOADER = yes;
+      SND_HDA_CODEC_CA0132_DSP = yes; # Enable DSP firmware loading on Creative Soundblaster Z/Zx/ZxR/Recon
```
Comment (flokli) [inline pkgs/os-specific/linux/kernel/common-config.nix]: This should link to some issue / commit upstream, so it's possible to check when we can drop it.

Comment (Emantor) [reply pkgs/os-specific/linux/kernel/common-config.nix]: Looking at Takshis tree this is staged for next, which will mean this will be in 5.8. I'll amend.

**Comment by flokli** [comment]
Sorry for the delay. Tested building `linux` and `linux_latest`.

**Comment by flokli** [comment]
So, this merged into staging, but GitHub seems to have hiccups.

**Pull request closed without merging**